Source/Scene/Cesium3DTilesetTraversal.js

```diff
@@ -1,856 +1,673 @@
-            if (replacementWithContent) {
-                if (!child.hasEmptyContent) {
-                    allReady = allReady && child.contentAvailable;
-                } else {
-                    allReady = allReady && this.internalDFS.execute(child);
-                }
+    function selectDesiredTile(tileset, tile, frameState) {
+        if (!skipLevelOfDetail(tileset)) {
+            if (tile.contentAvailable) {
+                // The tile can be selected right away and does not require traverseAndSelect
+                selectTile(tileset, tile, frameState);
+                tile._finalResolution = true;
             }
+            return;
         }
 
-        if (allReady) {
-            return children;
+        // If this tile is not loaded attempt to select its ancestor instead
+        var loadedTile = tile.contentAvailable ? tile : tile._ancestorWithContentAvailable;
+        if (defined(loadedTile)) {
+            // Tiles will actually be selected in traverseAndSelect
+            loadedTile._shouldSelect = true;
+        } else {
+            // If no ancestors are ready traverse down and select tiles to minimize empty regions.
+            // This happens often for immediatelyLoadDesiredLevelOfDetail where parent tiles are not necessarily loaded before zooming out.
+            selectDescendants(tileset, tile, frameState);
         }
+    }
 
-        return emptyArray;
-    };
-
-    function baseUpdateAndCheckChildren(tileset, tile, baseScreenSpaceError, frameState) {
-        if (hasAdditiveContent(tile)) {
-            tileset._desiredTiles.push(tile);
-        }
+    function visitTile(tileset) {
+        ++tileset._statistics.visited;
+    }
 
-        // Stop traversal on the subtree since it will be destroyed
-        if (tile.hasTilesetContent && tile.contentExpired) {
-            return false;
+    function touchTile(tileset, tile, frameState) {
+        if (frameState.passes.pick) {
+            return;
         }
+        tileset._cache.touch(tile);
+        tile._touchedFrame = frameState.frameNumber;
+    }
 
-        // stop traversal when we've attained the desired level of error
-        if (tile._screenSpaceError <= baseScreenSpaceError && !tile.hasTilesetContent) {
-            // update children so the leaf handler can check if any are visible for the children union bound optimization
-            updateChildren(tile, frameState);
-            return false;
+    function getPriority(tileset, tile) {
+        // If skipLevelOfDetail is off we try to load child tiles as soon as possible so that their parent can refine sooner.
+        // Additive tiles always load based on distance because it subjectively looks better.
+        // There may be issues with mixing additive and replacement tiles since SSE and distance are different types of values.
+        // Maybe all priorities need to be normalized to 0-1 range.
```

Comment:
Should this be part of this PR?

Reply:
If it's not a common case, I would say future work.

Reply:
Added to #6243 

>  How can we mix and match different prioritization modes. Additive tiles use distance, replacement tiles use SSE, terrain uses distance but should probably use SSE. Is there any way to normalize all values to a 0-1 range?